### ISSUE TITLE
bug: can't switch "write_mode" between "overwrite-delete-stale", "overwrite", "append"

### ISSUE THREAD

**Issue by ozbillwang** [cloudquery/cloudquery#12526]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

I have run the sync successfully, but after change the `write_mode` from "overwrite-delete-stale" to "append" , or "append" to "overwrite-delete-stale", the sync command will throw errors. 

1. If not set migrate_mode, it recommends to enable `migrate_mode `to `forced`

2. after set `migrate_mode `to `forced`, it starts to give errors as below

```
\ Syncing resources... (56/-, 186 resources/s) [0s] Error: failed to sync v3 source azure: write client returned error (insert): plugin returned error: failed to drop tabl
e azure_windowsiot_services: ERROR: cannot drop table azure_windowsiot_services because other objects depend on it (SQLSTATE 2BP01)
```
follow the instruction, I drop that table
```
drop table azure_windowsiot_services CASCADE;
```
it will ask to drop another one, no end. 
```
| Syncing resources... (59/-, 17 resources/s) [4s] Error: failed to sync v3 source azure: write client returned error (insert): plugin returned error: failed to drop table
 azure_policy_assignments: ERROR: cannot drop table azure_policy_assignments because other objects depend on it (SQLSTATE 2BP01)
\ Syncing resources... (59/-, 15 resources/s) [4s]
```
Finally I have to drop all tables with `CASCADE`, but that's not correct way, right?

Got similar issue for both AWS and Azure

### Expected Behavior

should be no issue

### CloudQuery (redacted) config

```
$ cat postgresql.yml 

kind: destination
spec:
  name: "postgresql"
  path: "cloudquery/postgresql"
  version: "v5.0.1"
  #write_mode: "append"
  write_mode: "overwrite-delete-stale"
  migrate_mode: forced # recommended, when upgrade with new version of plugins
  spec:
    connection_string: ""
```

### Steps To Reproduce

1. run sync command, but switch "write_mode" between "overwrite-delete-stale", "overwrite", "append"

### CloudQuery (redacted) logs

```
\ Syncing resources... (56/-, 186 resources/s) [0s] Error: failed to sync v3 source azure: write client returned error (insert): plugin returned error: failed to drop tabl
e azure_windowsiot_services: ERROR: cannot drop table azure_windowsiot_services because other objects depend on it (SQLSTATE 2BP01)
```

### CloudQuery version

cloudquery version 3.10.1

### Pull request (optional)

- [ ] I can submit a pull request

**Comment by erezrokah** [comment]
Hi @ozbillwang, thanks for opening the issue. The behavior you're seeing is expected and I'll split my explanation into two:

---

### 1. Switching from `overwrite-delete-stale` to `append` requires forced migration

The reason for that is when you configure `write_mode: append` we remove any primary keys configured on the tables, which generates a new schema for each table, that we don't support migrating automatically yet.

Similar to ⬆️  When you switch from `write_mode: append` to `write_mode: overwrite-delete-stale` we try to add the relevant primary keys to each table, which we don't support migrating automatically yet.

**Hence the need for a `migrate_mode: forced` will drops the tables.**

Supporting the former might be easier, as we could try and drop the constraints, but the latter could be more difficult as we'll have to handle duplicates if exists. In both cases the operations can be time consuming and stress the database (maybe `migrate_mode: best_effort_but_slow_and_can_fail` or similar is what we need).

---

### 2. `migrate_mode: forced` fails to drop the table

During `migrate_mode: forced` we use a simple `drop table` query, and don't `CASCADE`, as we don't want to destroy any dependent objects (e.g. views) the user created.

---

**TLDR**: It's currently not supported to switch between `overwrite-delete-stale` to `append` (and `append` to `overwrite-delete-stale`), unless you drop the database.

My suggestion is to:
1.  Add a note about `write_mode` in the [relevant docs](https://www.cloudquery.io/docs/reference/destination-spec#write_mode)
2. Add a note about `migrate_mode: forced` that it could fail if there are user created dependent objects
3. Make this issue a feature request to support switching between modes.

Please let me know what you think

**Comment by ozbillwang** [comment]
Thanks for the explanation
